static/js/components/search/NestedAggregatingMenuFilter.js

```diff
@@ -0,0 +1,292 @@
+import PatchedMenuFilter from './PatchedMenuFilter';
+import {
+  FacetAccessor, TermQuery, NestedQuery, TermsBucket, FilterBucket,
+  BoolMust, AggsContainer, CardinalityMetric
+} from "searchkit";
+import _ from 'lodash';
+import R from 'ramda';
+
+const REVERSE_NESTED_AGG_KEY = 'top_level_doc_count';
+const INNER_TERMS_AGG_KEY = 'nested_terms';
+
+/**
+ * Produces a Searchkit TermsBucket that includes a "reverse nested" aggregation.
+ *
+ * Example return value: {
+ *   "nested_terms": {
+ *     "terms":{
+ *       "field":"program.enrollments.course_title", ...
+ *     },
+ *     "aggs":{
+ *       "top_level_doc_count":{"reverse_nested":{}}
+ *     }
+ *   }
+ * }
+ */
+function ReverseNestedTermsBucket(key, field, options) {
+  let reverseNestedAgg = AggsContainer(REVERSE_NESTED_AGG_KEY, {'reverse_nested': {}});
+  return TermsBucket(key, field, options, reverseNestedAgg);
+}
+
+class NestedAggregatingFacetAccessor extends FacetAccessor {
+  /**
+   * Overrides buildSharedQuery in FacetAccessor
+   * If any filters are applied on the nested documents at this path, we need to build the shared
+   * query as normal, then alter the values for some searchkit internals in order to produce
+   * the correct query for ES. Searchkit essentially treats multiple filters on a nested set of
+   * documents as 'OR' filters, and the changes to 'query.index.filters' and 'query.index.filtersMap'
+   * are being made in order to treat those filters as 'AND'.
+   */
+  buildSharedQuery(query) {
+    query = super.buildSharedQuery(query);
+
+    // Construct an 'AND' query for all filters applied on this nested path
+    let groupedNestedFilter = this.createGroupedNestedFilter(query);
+    if (groupedNestedFilter) {
+      let nestedPath = this.getNestedPath();
+      // Find the element that Searchkit added to query.index.filters for this nested document and remove it
+      _.remove(query.index.filters, (filter) => (
```

Comment:
We need to change the values that searchkit sets for `query.index.filters` and `query.index.filtersMap` in order for the filters to work the way we want them to. The comment on this function (`buildSharedQuery`) has a little detail about it